what_url/ada_adapter.py

```diff
@@ -0,0 +1,199 @@
+from what_url._ada_wrapper import ffi, lib
+
+URL_ATTRIBUTES = (
+    'href',
+    'username',
+    'password',
+    'protocol',
+    'port',
+    'hostname',
+    'host',
+    'pathname',
+    'search',
+    'hash',
+)
+PARSE_ATTRIBUTES = URL_ATTRIBUTES + ('origin',)
+
+
+def _get_str(x):
+    ret = ffi.string(x.data, x.length).decode('utf-8') if x.length else ''
+    return ret
+
+
+def check_url(s):
```

Comment:
Can you make sure that the function names are aligned with the C/Rust implementation? For example, this needs to be `can_parse`. Also, there might be a second parameter to this function.